parser/parser.go

```diff
@@ -538,52 +538,57 @@ func (p *parser) parseObjectRemainder(tok *token) (ast.Node, *token, error) {
 
 /* parses for x in expr for y in expr if expr for z in expr ... */
 func (p *parser) parseComprehensionSpecs(end tokenKind) (*ast.ForSpec, *token, error) {
-	var ifSpecs []ast.IfSpec
+	var parseComprehensionSpecsHelper func() (*ast.ForSpec, *ast.ForSpec, *token, error)
+	parseComprehensionSpecsHelper = func() (*ast.ForSpec, *ast.ForSpec, *token, error) {
```

Comment:
if you pass in 'outer' (can be nil) then you can drop the extra return param and do it in the ForSpec constructor, which is a bit simpler overall.